tests/backends/aiida_sqlalchemy/test_migrations.py

```diff
@@ -779,7 +779,7 @@ def setUpBeforeMigration(self):
                 param_data = session.query(DbLog).filter(DbLog.objpk == param.id
                                                          ).filter(DbLog.objname == 'something.else.'
                                                                   ).with_entities(*cols_to_project).one()
-                serialized_param_data = dumps_json([(dict(list(zip(param_data.keys(), param_data))))])
+                serialized_param_data = dumps_json([param_data._asdict()])
```

Comment:
I mean, I know it is more compact and simple, but should we be using private methods of an sqla object?? Is this how they intend you to do it? Why don't they make it a public method then?

Reply:
Its definitely how they intend it to be used (the warning tells you to use it). Its like us an `AttrDict`s; the attributes are intended to primarily be the keys

Reply:
Ok, I see. I still find it bizarre that they are suggesting users of the object to access some functionality through a private method. I mean, what makes it "private" then?

Reply:
Well exactly, thats why personally I think it is never good to be using "attribute dictionaries" in Python (as opposed to e.g. JavaScript where it is a feature of the language)

Reply:
Mmm I don't understand. My objection was more one of intentionality (as in, the point of prefacing your methods with the underscore is to say "I am using this internally, this is not meant for you, use it at your own peril of messing the object up / losing the feature without deprecation").

In that sense attribute dictionaries are in principle intentionally exposed, what would be the problem with that?

(I mean, I have my own quarrel with attribute dicts when making workflows and crashing against some unexpected problems of where exactly the attribute dict starts having simple dicts, but other than that I recognize it is convenient for the shell)

Reply:
because if all the attributes on your class are meant to (dynamically) relate to keys of a dictionary, then you cannot have any "static" attributes/methods associated with your class. Here they have "compromised" and have any static methods public but underscored.
The better route would probably have been to use an "external" function (that would call the private methods), as e.g. you do with https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict